src/sources/socket/mod.rs

```diff
@@ -279,35 +279,32 @@ impl SourceConfig for SocketConfig {
                     )
             }
             Mode::Udp(config) => {
-                let host_key_path = config
+                let legacy_host_key = config
                     .host_key()
                     .as_ref()
-                    .and_then(|x| parse_value_path(x).ok())
                     .map_or_else(
-                        || LegacyKey::InsertIfEmpty(owned_value_path!("host")),
-                        LegacyKey::InsertIfEmpty,
-                    );
+                        || parse_value_path(log_schema().host_key()).ok(),
+                        |k| k.path.clone(),
+                    )
+                    .map(LegacyKey::InsertIfEmpty);
 
-                let port_key_path = config
+                let legacy_port_key = config
                     .port_key()
                     .as_ref()
-                    .and_then(|x| parse_value_path(x).ok())
-                    .map_or_else(
-                        || LegacyKey::InsertIfEmpty(owned_value_path!("port")),
-                        LegacyKey::InsertIfEmpty,
-                    );
+                    .map_or_else(|| parse_value_path("path").ok(), |k| k.path.clone())
```

Comment:
Should this be 

```suggestion
                    .map_or_else(|| parse_value_path("port").ok(), |k| k.path.clone())
```

Reply:
Fixed in [17c7645](https://github.com/vectordotdev/vector/pull/15666/commits/17c76453a258342a101933f934fe8c7c1af5401f)